k8s/src/main/scala/io/buoyant/k8s/EventLogging.scala

```diff
@@ -1,55 +1,95 @@
 package io.buoyant.k8s
 
+import com.twitter.logging.Logger
+
 private[k8s] trait EventLogging {
   def ns: String
   def srv: String
-
-  @inline private[this] def logAction[A](verb: String, descriptor: => String, format: A => String)(value: A): Unit =
+  
+  @inline private[this] def logAction[A](verb: String, descriptor: String, format: A => String)(value: A): Unit =
     log.ifTrace(s"k8s ns $ns service $srv $verb $descriptor ${format(value)}")
 
-  protected def logActions[A](verb: String, descriptor: => String, format: A => String)(values: Iterable[A]): Unit =
-    if (values.nonEmpty) {
-      log.debug(
-        "k8s ns %s service %s %s %ss",
-        ns, srv, verb, descriptor
-      )
-      values.foreach(logAction(verb, descriptor, format))
-    }
-
-  private[this] def formatMapping[A, B](kv: (A, B)): String = kv match {
+  @inline private[this] def formatMapping[A, B](kv: (A, B)): String = kv match {
     case (name: String, port: Int) => s"'$name' to port $port"
     case (name: String, to) => s"'$name' to $to"
     case (fromPort, to) => s"from $fromPort to $to"
   }
 
-  @inline def deletion[A, B](deleted: Map[A, B]): Unit =
-    logActions("deleted", "port mapping", formatMapping)(deleted)
+  @inline private[this] def defaultFormat[A](a: A): String = a.toString
 
-  @inline def addition[A, B](added: Map[A, B]): Unit =
-    logActions("added", "port mapping", formatMapping)(added)
+  protected def mkDeletion[A](
+    descriptor: String,
+    format: A => String = defaultFormat[A](_)
+  )(
+    deleted: Iterable[A]
+  ): Unit =
+    if (deleted.nonEmpty) {
+      log.debug(
+        "k8s ns %s service %s deleted %ss",
+        ns, srv, descriptor
+      )
+      if (log.isLoggable(Logger.TRACE)) {
+        deleted.foreach(logAction("deleted", descriptor, format))
+      }
+    }
 
-  protected def logModification[A](descriptor: String)(mods: Iterable[(A, A)]): Unit =
-    if (mods.nonEmpty) {
+  def deletion[A, B](deleted: Map[A, B]): Unit =
+    mkDeletion("port mapping", formatMapping)(deleted)
+
+  protected def mkNewState[A](
+    descriptor: String,
+    format: A => String = defaultFormat[A](_)
```

Comment:
I think this is always `toString` (either explicitly with `_.toString` or by default with the default formatter.  Can we just remove it?